run_test.go

```diff
@@ -184,6 +185,64 @@ func TestRunsCancel(t *testing.T) {
 	})
 }
 
+func TestRunsForceCancel(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	wTest, wTestCleanup := createWorkspace(t, client, nil)
+	defer wTestCleanup()
+
+	// We need to create 2 runs here. The first run will automatically
+	// be planned so that one cannot be cancelled. The second one will
+	// be pending until the first one is confirmed or discarded, so we
+	// can cancel that one.
+	_, _ = createRun(t, client, wTest)
+	rTest, _ := createRun(t, client, wTest)
+
+	t.Run("run is not force-cancelable", func(t *testing.T) {
+		assert.False(t, rTest.Actions.IsForceCancelable)
+	})
+
+	t.Run("user is allowed to force-cancel", func(t *testing.T) {
+		assert.True(t, rTest.Permissions.CanForceCancel)
+	})
+
+	t.Run("after a normal cancel", func(t *testing.T) {
+		// Request the normal cancel
+		err := client.Runs.Cancel(ctx, rTest.ID, RunCancelOptions{})
+		assert.NoError(t, err)
+
+		// Allow time for the async cancel request to operate.
+		time.Sleep(5 * time.Second)
```

Comment:
This looks like something that could end up causing flaky tests. In all the `helper_test.go` cases where I would need a certain state I just created a small loop and checked for the actual state I needed before moving on. Its a few more lines of code but think it would be a bit more robust to use that pattern here as well: https://github.com/hashicorp/go-tfe/blob/master/helper_test.go#L302-L318